T/Torch/Torch/build_tarballs.jl

```diff
@@ -0,0 +1,267 @@
+        -DCAFFE2_CUSTOM_PROTOC_EXECUTABLE=$host_bindir/protoc \
+        -Wno-dev \
+        $cmake_extra_args \
+        ..
+}
+if [[ $bb_full_target != *cuda* ]]; then
+    configure
+else
+    configure
+    configure || configure
+fi
+cmake --build . -- -j $nproc
+make install
+install_license ../LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+filter!(p -> !(Sys.islinux(p) && libc(p) == "musl"), platforms) # musl fails due to conflicting declaration of C function ‘void __assert_fail(const char*, const char*, int, const char*) - between /opt/x86_64-linux-musl/x86_64-linux-musl/include/c++/8.1.0/cassert:44 and /opt/x86_64-linux-musl/x86_64-linux-musl/sys-root/usr/include/assert.h
+filter!(!Sys.iswindows, platforms) # ONNX does not support cross-compiling for w64-mingw32 on linux
+filter!(p -> arch(p) != "powerpc64le", platforms) # PowerPC64LE is not supported by XNNPACK
+filter!(p -> !(Sys.isapple(p) && arch(p) == "aarch64"), platforms) # aarch64-apple not supported by CPUInfo_jll v0.0.20200612 referenced by XNNPACK_jll v0.0.20200323
+filter!(!Sys.isfreebsd, platforms) # Build fails: Clang v12 crashes compiling aten/src/ATen/native/cpu/MaxUnpoolKernel.cpp.
+
+mkl_platforms = [
+    Platform("x86_64", "Linux"),
+    Platform("i686", "Linux"),
+    Platform("x86_64", "MacOS"),
+    Platform("x86_64", "Windows"),
+]
+
+blis_platforms = filter(p -> p ∉ mkl_platforms, [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("x86_64", "freebsd"),
+    Platform("aarch64", "macos"),
+    Platform("x86_64", "macos"),
+    Platform("x86_64", "windows"),
+])
+
+openblas_platforms = filter(p -> p ∉ union(mkl_platforms, blis_platforms), platforms)
+
+cuda_platforms = [
+    Platform("x86_64", "Linux"; cuda = "10.2"),
+    Platform("x86_64", "Linux"; cuda = "11.3"),
+]
+for p in cuda_platforms
+    push!(platforms, p)
+end
+
+# The products that we will ensure are always built
+products = [
+    LibraryProduct(["libtorch", "torch"], :libtorch),
+    LibraryProduct(["libtorch_cpu", "torch_cpu"], :libtorch_cpu),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae")),
+    Dependency("blis_jll"; platforms = blis_platforms),
+    Dependency("CPUInfo_jll"),
+    Dependency("CUDNN_jll"; platforms = cuda_platforms),
+    Dependency("Gloo_jll"; platforms = filter(p -> nbits(p) == 64, platforms)),
+    Dependency("LAPACK_jll"; platforms = openblas_platforms),
+    Dependency("MKL_jll"; platforms = mkl_platforms),
+    Dependency("MKL_Headers_jll"; platforms = mkl_platforms),
```

Comment:
Headers are build dependencies :slightly_smiling_face: 
```suggestion
    BuildDependency("MKL_Headers_jll"; platforms = mkl_platforms),
```